pkg/controller/external_nofork.go

```diff
@@ -313,27 +280,61 @@ func filterInitExclusiveDiffs(tr resource.Terraformed, instanceDiff *tf.Instance
 	if err != nil {
 		return errors.Wrap(err, "cannot get spec.initProvider parameters")
 	}
-	initProviderExclusiveParamKeys := getTerraformIgnoreChanges(paramsForProvider, paramsInitProvider)
 
+	initProviderExclusiveParamKeys := getTerraformIgnoreChanges(paramsForProvider, paramsInitProvider)
 	for _, keyToIgnore := range initProviderExclusiveParamKeys {
 		for attributeKey := range instanceDiff.Attributes {
-			if keyToIgnore != attributeKey {
+			keyToIgnoreAsPrefix := fmt.Sprintf("%s.", keyToIgnore)
+			if keyToIgnore != attributeKey && !strings.HasPrefix(attributeKey, keyToIgnoreAsPrefix) {
 				continue
 			}
 
-			if err := deleteInstanceDiffAttribute(instanceDiff, keyToIgnore); err != nil {
-				return errors.Wrapf(err, "cannot delete key %q from instance diff", keyToIgnore)
-			}
+			delete(instanceDiff.Attributes, attributeKey)
 
-			keyComponents := strings.Split(keyToIgnore, ".")
+			// TODO: tags-tags_all implementation is AWS specific.
+			// Consider making this logic independent of provider.
+			keyComponents := strings.Split(attributeKey, ".")
 			if keyComponents[0] != "tags" {
```

Comment:
Yes, we will need to consider this also while initializing the `tags_all` argument. These have to be provider specific.

Reply:
I copied to-do comment to all places where we duplicate the logic, to make sure that we don't miss any when addressing concerns.